cpmpy/solvers/choco.py

```diff
@@ -0,0 +1,617 @@
+
+        # translate solution values (of user specified variables only)
+        self.objective_value_ = None
+        if has_sol:
+            # fill in variable values
+            for cpm_var in self.user_vars:
+                value = sol.get_int_val(self.solver_var(cpm_var))
+                if isinstance(cpm_var, _BoolVarImpl):
+                    cpm_var._value = bool(value)
+                else:
+                    cpm_var._value = value
+
+            # translate objective
+            if self.has_objective():
+                self.objective_value_ = sol.get_int_val(self.solver_var(self.obj))
+
+        return has_sol
+
+    def solveAll(self, display=None, time_limit=None, solution_limit=None, **kwargs):
+        """
+            Compute all (optimal) solutions, map them to CPMpy and optionally display the solutions.
+
+            Arguments:
+                - display: either a list of CPMpy expressions, OR a callback function, called with the variables after value-mapping
+                        default/None: nothing displayed
+                - solution_limit: stop after this many solutions (default: None)
+                - time_limit:  maximum solve time in seconds (float, default: None)
+
+            Returns: number of solutions found
+        """
+
+        if time_limit is not None:
+            self.chc_solver.limit_time(str(time_limit) + "s")
+
+        self.chc_solver = self.chc_model.get_solver()
+        start = time.time()
+        if self.has_objective():
+            sols = self.chc_solver.find_all_optimal_solutions(maximize=self.maximize_obj,
+                                                                         solution_limit=solution_limit,
+                                                                         objective=self.solver_var(self.obj))
+        else:
+            sols = self.chc_solver.find_all_solutions(solution_limit=solution_limit)
+        end = time.time()
+
+        # new status, get runtime
+        self.cpm_status = SolverStatus(self.name)
+        self.cpm_status.runtime = end - start
+
+        # display if needed
+        if display is not None:
+            for sol in sols:
+                # map the solution to user vars
+                for cpm_var in self.user_vars:
+                    value = sol.get_int_val(self.solver_var(cpm_var))
+                    if isinstance(cpm_var, _BoolVarImpl):
+                        cpm_var._value = bool(value)
+                    else:
+                        cpm_var._value = value
+                # print the desired display
+                if isinstance(display, Expression):
+                    print(display.value())
+                elif isinstance(display, list):
+                    print([v.value() for v in display])
+                else:
+                    display()  # callback
+
+        return len(sols)
+
+    def solver_var(self, cpm_var):
+        """
+            Creates solver variable for cpmpy variable
```

Comment:
So this one allows integers to not be converted to a choco var, but i think there is still a problem with np.integers not being supported in choco, so I guess those have to be converted here to a python int